File: DHealth/app/src/main/java/com/example/hamid/dhealth/Utils/ImageUtils.java
package com.example.hamid.dhealth.Utils;

import android.content.Context;
import android.graphics.Bitmap;

public class ImageUtils {


    public static Bitmap scaleDownBitmap(Bitmap photo, Context context) {

        int newHeight = 100;

        final float densityMultiplier = context.getResources().getDisplayMetrics().density;

        int h = (int) (newHeight * densityMultiplier);
        int w = (int) (h * photo.getWidth() / ((double) photo.getHeight()));

        photo = Bitmap.createScaledBitmap(photo, w, h, true);

        return photo;
    }


}

File: DHealth-Android/Kotlin/app/src/main/java/com/example/hamid/dhealth/utils/ImageUtils.kt
package com.example.hamid.dhealth.utils

import android.content.Context
import android.graphics.Bitmap

object ImageUtils {


    fun scaleDownBitmap(photo: Bitmap, context: Context): Bitmap {
        var photo = photo

        val newHeight = 100

        val densityMultiplier = context.resources.displayMetrics.density

        val h = (newHeight * densityMultiplier).toInt()
        val w = (h * photo.width / photo.height.toDouble()).toInt()

        photo = Bitmap.createScaledBitmap(photo, w, h, true)

        return photo
    }


}
